src/AzAcme.Cli/Commands/OrderCommand.cs

```
@@ -64,47 +65,59 @@ protected override async Task<int> OnExecute(OrderOptions opts)
             
             this.logger.LogInformation("Updating DNS challenge records.");
             await dnsZone.SetTxtRecords(order);
+            try {
 
-            // allow up to 1 minute
-            var attempts = 12;
-            var delaySeconds = 5;
+                // allow up to 1 minute
+                var attempts = 12;
+                var delaySeconds = 5;
 
-            this.logger.LogInformation("Waiting for DNS records to be verified.");
+                if (opts.PropDelay != 0)

```

Thanks for the contribution, could we change to make define the absolute time we're will to wait, and calculate the attempts based on that. 

optional param: verficiation-timeout (default to 60).

The waiting in then already handled in the existng functionality and removes the Thread wait here

---

I'm by no means really knowledgeable about ACME.  But let me elaborate on my observations about the behavior I was encountering.

First the DNS challenge was setup in azure DNS, then the validation loop would run, the first pass would fail with NXDOMAIN in the error response from letsencrypt and the c# routine parsing the response would throw an exception.  So I suspect the attempt to re-try in the validation routine isn't always working.  

I also was suspicious that sometimes an exception would be thrown on a second attempt due to letsencrypt doing something stateful on the first attempt that invalidated the challenge.

So while a bit of a hack... simply adding some delay before validating the challenge seemed to give azure-DNS enough time that it would respond to the DNS query from letsencrypt.  6 seconds or so seemed to work but in my automation I'm using 30.  But the main thing was that the existing waits happened _after_ the point in time that I seemed to need to wait to be successful.

---

Ah ok, is there an exception / error you experienced you can add as an issue, so can see the location / what's throwing it?

Ideally I'd like to fix the bit that's throwing the exception over adding additional wait time, so it can swallow that specific error scenario in the existing wait loop and keep trying with the validation since it's a 'known' and 'valid' error scenario to ignore and check status again next loop?

I'm assuming it's erroring in the validation loop? It certainly supports multiple checks (sometimes takes up to 15-20 seconds for me, so that's 2-3 loops) so bit odd, if you can supply error it would be great.

Good point on the finally, either way will be fine.

---

Anyways... I really appreciate you putting this tool together and making it available, very much resolved a problem for me.  If there is a better way to fix this.... all good - I opened an issue.  My knowledge of acme was pretty much already tapped out here.

---

FWIW, I finally updated to the current version of az-acme... and I'm still seeing this problem.  I modified the source locally to just insert a single sleep for 30 seconds before  await this.WaitForVerificationWithTable(order, acmeDirectory, attempts, delaySeconds, opts); and... works reliably.  My original theory still seems to fit - azure dns seems to be eventual consistency coherent at least within the resolution of a few seconds, so the local dns client that you run az-acme on can see the txt record before other hosts can... as such lets encrypt when it is asked to validate the challenge response within the coherency window gets a NX response and invalidates the challenge.